### PR TITLE
Fix for SSO Login Issue

### PR DESCRIPTION
**Pull request by WadeBarnes**
Refactor BuildUrlString to use UriBuilder to generate complete URL fo…

- Updated callers to reference the new function signature
- Added unit tests and Bogus library to make it easier to generate random test data

- From https://github.com/bcgov/jasper/pull/20

### CODE REVIEW

File: api/helpers/XForwardedForHelper.cs [hunk 1]
```diff
@@ -1,11 +1,38 @@
-﻿namespace SS.Api.helpers
+﻿using System;
+
+namespace Scv.Api.Helpers
```
Comment (sturufous) [inline api/helpers/XForwardedForHelper.cs:3]: Should this be referencing Scv?

File: api/helpers/XForwardedForHelper.cs [hunk 2]
```diff
@@ -1,11 +1,38 @@
...
+            var uriBuilder = new UriBuilder
+            {
+                Scheme = "https",
+                Host = forwardedHost,
+                Path = sanitizedPath,
+                Query = query
+            };
+
+            var portComponent =
+                string.IsNullOrEmpty(forwardedPort) || forwardedPort == "80" || forwardedPort == "443"
```
Comment (sturufous) [inline api/helpers/XForwardedForHelper.cs:24]: I added port 8080 here for scv to support running on Emerald.

File: api/helpers/XForwardedForHelper.cs (hunk 1 above)
Comment (WadeBarnes) [reply api/helpers/XForwardedForHelper.cs]: Yup, copy paste error.

File: api/helpers/XForwardedForHelper.cs (hunk 2 above)
Comment (WadeBarnes) [reply api/helpers/XForwardedForHelper.cs:24]: Purposely not doing that to keep things as similar to what's in `prod` `silver` now.

**Review by sturufous — APPROVED** [review]

**Pull request merged**